### PR TITLE
Made some changes based on suggestions and looking through code and finding a few actual issues

### PR DESCRIPTION
**Pull request by amhester**

### CODE REVIEW

File: src/Francis.js [hunk 1]
```diff
@@ -119,31 +120,42 @@ class Francis {
...
-        requestsToAbort.forEach(request => request.xhr.abort());
-        //remove all requests related to that transaction from the working set
-        self.workingRequests = self.workingRequests.filter(request => request.transactionId !== transactionId);
+
+        //What are your thoughts on this change, should minimize the amount of looping over stuff.
+        for(let i = 0; i < self.workingRequests.length; i++) {
+            let workItem = self.workingRequests[i];
+            if(workItem.transactionId === transactionId) {
+                workItem.xhr.abort();
+                self.workingRequests.splice(i,1);
```
Comment (willcj33) [inline src/Francis.js]: i don't think you can do this mid loop can you? Will it change the indexes? we need to have an else probably that just adds to a temp array and reassigns the global array to the temp at the end.

File: src/Francis.js [hunk 2]
```diff
@@ -227,8 +239,9 @@ class Francis {
             } else if (transaction.status === Francis.QUERIER_STATUS.LOCKED || transaction.status === Francis.QUERIER_STATUS.ABORTING) {
                 ///TODO: probably just do nothing
             } else {
-                ///I think I'm done checking shit for other cases... for now
-                self.workQueue.push(transaction.actions.shift());
+                if(transaction.mode === Francis.QUERIER_MODE.SINGLE) {
```
Comment (willcj33) [inline src/Francis.js:230]: good call

File: src/Francis.js (hunk 1 above)
Comment (amhester) [inline src/Francis.js]: You're right, should just keep track of them and then execute splicing at the end.

**Comment by amhester** [comment]
Ok, going to merge for now, lets get back to master and go from there.

**Pull request merged**